subsys/net/lib/http/http_server.c

```diff
@@ -0,0 +1,1038 @@
+
+	const char *method;
+	const char *path;
+
+	if (!has_upgrade_header) {
+		enum http_method method_h2c = parser.method;
+
+		method = http_method_str(method_h2c);
+		path = url_buffer;
+	} else {
+		method = header_parsing(ctx_client, HTTP_HPACK_METHOD);
+		path = header_parsing(ctx_client, HTTP_HPACK_PATH);
+	}
+
+	if (strcmp(method, "GET") == 0 && strcmp(path, "/") == 0) {
+
+		generate_response_headers_frame(response_headers_frame, frame->stream_identifier,
+						HTTP_HPACK_STATUS_200);
+
+		size_t ret = sendall(ctx_client->client_fd, response_headers_frame,
+				     sizeof(response_headers_frame));
+		if (ret < 0) {
+			LOG_ERR("ERROR writing to socket");
+			return -errno;
+		}
+
+		size_t content_size = sizeof(content_200);
+
+		send_data(ctx_client->client_fd, content_200, content_size, HTTP_DATA_FRAME,
+			  HTTP_FLAG_END_STREAM, frame->stream_identifier);
+	} else {
+		generate_response_headers_frame(response_headers_frame, frame->stream_identifier,
+						HTTP_HPACK_STATUS_404);
+
+		size_t ret = sendall(ctx_client->client_fd, response_headers_frame,
+				     sizeof(response_headers_frame));
+		if (ret < 0) {
+			LOG_ERR("ERROR writing to socket");
+			return -errno;
+		}
+
+		size_t content_size = sizeof(content_404);
+
+		send_data(ctx_client->client_fd, content_404, content_size, HTTP_DATA_FRAME,
+			  HTTP_FLAG_END_STREAM, frame->stream_identifier);
+	}
+
+	ctx_client->server_state = HTTP_FRAME_HEADER_STATE;
+
+	return 0;
+}
+
+int handle_http_frame_priority(struct http_client_ctx *ctx_client)
+{
+	LOG_DBG("HTTP_FRAME_PRIORITY_STATE");
+
+	struct http_frame *frame = &ctx_client->current_frame;
+
+	print_http_frames(frame);
+
+	ctx_client->server_state = HTTP_FRAME_HEADER_STATE;
+
+	return 0;
+}
+
+int handle_http_frame_continuation(struct http_client_ctx *ctx_client)
+{
+	LOG_DBG("HTTP_FRAME_CONTINUATION_STATE");
+	ctx_client->server_state = HTTP_FRAME_HEADERS_STATE;
+
+	return 0;
+}
+
+int handle_http_frame_settings(struct http_client_ctx *ctx_client)
+{
+	LOG_DBG("HTTP_FRAME_SETTINGS");
+
+	struct http_frame *frame = &ctx_client->current_frame;
+
+	print_http_frames(frame);
+
+	if (!settings_ack_flag(frame->flags)) {
+		ssize_t ret =
+			sendall(ctx_client->client_fd, settings_frame, sizeof(settings_frame));
+		if (ret < 0) {
+			LOG_ERR("ERROR writing to socket");
+			return -errno;
+		}
+
+		ret = sendall(ctx_client->client_fd, settings_ack, sizeof(settings_ack));
+		if (ret < 0) {
+			LOG_ERR("ERROR writing to socket");
+			return -errno;
+		}
+	}
+
+	ctx_client->server_state = HTTP_FRAME_HEADER_STATE;
+
+	return 0;
+}
+
+int handle_http_frame_window_update(struct http_client_ctx *ctx_client)
+{
+	LOG_DBG("HTTP_FRAME_WINDOW_UPDATE");
+
+	struct http_frame *frame = &ctx_client->current_frame;
+
+	print_http_frames(frame);
+
+	if (!has_upgrade_header) {
+
+		frame->stream_identifier = 1;
+		handle_http_frame_headers(ctx_client);
+		ctx_client->server_state = HTTP_FRAME_GOAWAY_STATE;
+
+		return 0;
+	}
```

Comment:
I don't think I understand this part, WINDOW_UPDATE has nothing to do with headers.

Reply:
This part was for h2c upgrade, it was a special case because it initially sent me a request using the HTTP/1.1 protocol, then upgraded to HTTP/2.0. I dont receive headers frame from it that's why I had to handle it request in differrent way